### ISSUE TITLE
File is not shared with user whose username contains capital letter

### ISSUE THREAD

**Issue by HariBhandari07** [owncloud/core#35382]
### Steps to reproduce
1. Create a user with upper case username (eg. User1)
2. Share a file(eg somefile.txt) with User1 using webui
3. Relogin as User1
4. The shared file (somefile.txt) is not listed in the files page and shared with you page

### Expected behaviour
file should be shared with user with uppercase username
### Actual behaviour
file is not shared with user with uppercase username

### Server configuration
**Operating system**:
Ubuntu 18.04.2 LTS
**Web server:**
Apache
**Database:**
SQLite
**PHP version:**
7.1
**ownCloud version:** (see ownCloud admin page)
11.0.0.0
**Updated from an older ownCloud or fresh install:**
fresh install
**Where did you install ownCloud from:**
git

**List of activated apps:**

```
  - comments: 0.3.0
  - dav: 0.4.0
  - encryption: 1.3.1
  - federatedfilesharing: 0.5.0
  - federation: 0.1.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_sharing: 0.12.0
  - files_texteditor: 2.3.0
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - guests: 0.8.1
  - market: 0.4.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - testing: 0.1.0
  - updatenotification: 0.2.1
  - user_management: 0.0.1

```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"IZmafUYSKehiSWdPKnZr","level":3,"time":"2019-05-31T04:53:57+00:00","remoteAddr":"::1","user":"admin","app":"no app in context","method":"POST","url":"\/core\/index.php\/apps\/user_management\/users","message":"User backend OC\\User\\Database is returning home: \/home\/haribhandari\/www\/core\/data\/User1 for user: User1 which differs from existing value: "}
{"reqId":"FHbiNrBQKTnBrPVjtGaf","level":2,"time":"2019-05-31T04:54:03+00:00","remoteAddr":"::1","user":"admin","app":"no app in context","method":"GET","url":"\/core\/remote.php\/dav\/avatars\/admin\/31.jpeg","message":""}
```

**Comment by phil-davis** [comment]
The same issue happens whatever the display name of the user. e.g. create `User1` with display-name `John Smith`
```
php occ user:add --display-name "John Smith" User1
```

Then share a file or folder with "John Smith" on the webUI.

Login to the webUI as `User`. It shows the correct display-name `John Smith`.
But the share is not seen.

**Comment by davitol** [comment]
Tested in 10.2.0.5 and it is not reproducible, so the issue was included after oC 10.2 release and it is not in a released version (cc @micbar )

**Comment by phil-davis** [comment]
I confirmed by running the scenario in #35388 on top of the `release-10.2.0` branch.
So this is a "new" regression.
We will `git bisect`

**Comment by micbar** [comment]
maybe this comes from https://github.com/owncloud/core/pull/35084

**Comment by phil-davis** [comment]
`git bisect` shows that the issue happened with commit:
https://github.com/owncloud/core/commit/3e7bfb87c065c71bb88238275a5c84fb95947dc1
which is PR #35138 (stable10) which is the backport of #35084 

So the case for "ordinary" UID containing upper-case now does not work.

Needs a bit of thinking first. Why do we allow creating UIDs with uppercase in them?
```
php occ user:add SmithJ --displayname "John Smith"
```
The UID `SmithJ` actually acts the same as `smithj` to the API etc.
Maybe when adding a UID, we should just lowercase it in the first place?
Then we would always store UIDs lowercase, and so when we get a UID from somewhere (incoming API call, attempt at sharing in the webUI...) we can just always lowercase it and deal with it as lowercase in all backend code.

And we could do the same for UIDs coming from external authentication like LDAP. Because (I guess) if LDAP gives us a UID `SmithJ` then we can use `smithj` `SMITHJ` `SmithJ` when authenticating anyway.

**Comment by micbar** [comment]
@DeepDiver1975 
Please help to discuss it. This needs to be decided.

**Comment by micbar** [comment]
IMO this is a serious regression.
PR #35138 (stable10) and #35084 should be reverted.

**Comment by micbar** [comment]
@hodyroff @pmaier1 This should go into the hotfix, do you agree?

**Comment by sharidas** [comment]
Test on stable10 branch of core branch by creating a user with display name as suggested and here is the screencast: 
![uppercaseusername](https://user-images.githubusercontent.com/3600427/58776249-3be03600-85e7-11e9-87ae-7197a07bb1b2.gif)
Its not reproducible for me.

**Comment by micbar** [comment]
@sharidas It is reproducible with a username with mixed Case.

you tried "user1" --> works fine
try it with "User1" --> will fail

Than means, it will fail on UUIDs (LDAP)

**Comment by hodyroff** [comment]
> @hodyroff @pmaier1 This should go into the hotfix, do you agree?

Yes. Of course the original bug fix idea should also not be lost ...

**Comment by micbar** [comment]
@phil-davis  We should revert everything

Created PRs
https://github.com/owncloud/core/pull/35449

**Comment by phil-davis** [comment]
@micbar this issue is resolved by the reverts in PRs #35448 and #35449 

For hotfix release we will need to:
```
git checkout release-10.2.0
git checkout -b release-10.2.1
```
To make a branch for `10.2.1` that just starts directly from the released `10.2.0` code.
Then cherry-pick the commits from #35449 back to the new `release-10.2.1` branch.
(and also cherry-pick other stuff to go in `10.2.1`)

**Comment by micbar** [comment]
Fixed, needs new release.
